### DISCUSSION TITLE
Remove stack key from "E_ROUTE_NOT_FOUND" exception

### DISCUSSION THREAD

**Discussion by DZ8540** [adonisjs/core#4139] [Help] [answered]
Yo, i have handling exception like this

![telegram-cloud-photo-size-2-5366465378001209885-y](https://user-images.githubusercontent.com/50152205/235424211-7b2c4cf6-1766-4182-866d-b7cbe44d79ed.jpg)

With header `application/json` i have `stack` key in response's body.
How can i remove this `stack` key from response?

Thanks)

P.S.:
These are not worked for me:
- production mode (even via `node ace build --production`)
- https://docs.adonisjs.com/guides/exception-handling#disable-reporting-for-certain-exceptions

**Comment by RomainLanz** [comment, marked as answer]
Hey @DZ8540! 👋🏻 

It should disappear when you are in production.
You can also handle the exception yourself in the [exception handler](https://docs.adonisjs.com/guides/exception-handling#handling-exceptions-globally).

**Reply by DZ8540** [reply]
Thank you for response!

I forgot to write that I tried this before asking a question:
- production mode (even via `node ace build --production`)
- https://docs.adonisjs.com/guides/exception-handling#disable-reporting-for-certain-exceptions

All of these not worked for me(

**Reply by RomainLanz** [reply]
It is not about building for production but running the code in production mode.
You can do it by changing the `NODE_ENV`.

**Reply by DZ8540** [reply]
@RomainLanz thank you, problem was that, when i builded application via `node ace build --production`, i copied `.env` file from development folder. Oops)
